pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1049,6 +1124,25 @@ IdealState ensureAllPartitionsConsuming(TableConfig tableConfig, PartitionLevelS
     return idealState;
   }
 
+  private StreamPartitionMsgOffset getPartitionGroupStartOffset(StreamConfig streamConfig, int partitionGroupId) {
```

Comment:
suggest rename to getPartitionGroupEarliestOffset() Or, getPartitionGroupOldestOffset() or, perhaps smallestOffset